R/rdeephaven/R/table_handle_wrapper.R

```diff
@@ -1,147 +1,322 @@
+    },
+    
+    std_by = function(by = character()) {
+      verify_string("by", by, FALSE)
+      return(TableHandle$new(self$.internal_rcpp_object$std_by(by)))
+    },
+    
+    percentile_by = function(percentile, by = character()) {
+      verify_in_unit_interval("percentile", percentile, TRUE)
+      verify_string("by", by, FALSE)
+      return(TableHandle$new(self$.internal_rcpp_object$percentile_by(percentile, by)))
+    },
+    
+    count_by = function(col = "n", by = character()) {
+      verify_string("col", col, TRUE)
+      verify_string("by", by, FALSE)
+      return(TableHandle$new(self$.internal_rcpp_object$count_by(col, by)))
+    },
+    
+    cross_join = function(right_side, columns_to_match = character(), columns_to_add = character()) {
+      verify_string("columns_to_match", columns_to_match, FALSE)
+      verify_string("columns_to_add", columns_to_add, FALSE)
+      return(TableHandle$new(self$.internal_rcpp_object$cross_join(
+        right_side$.internal_rcpp_object,
+        columns_to_match, columns_to_add
+      )))
+    },
+    
+    natural_join = function(right_side, columns_to_match = character(), columns_to_add = character()) {
+      verify_string("columns_to_match", columns_to_match, FALSE)
+      verify_string("columns_to_add", columns_to_add, FALSE)
+      return(TableHandle$new(self$.internal_rcpp_object$natural_join(
+        right_side$.internal_rcpp_object,
+        columns_to_match, columns_to_add
+      )))
+    },
+    
+    exact_join = function(right_side, columns_to_match = character(), columns_to_add = character()) {
+      verify_string("columns_to_match", columns_to_match, FALSE)
+      verify_string("columns_to_add", columns_to_add, FALSE)
+      return(TableHandle$new(self$.internal_rcpp_object$exact_join(
+        right_side$.internal_rcpp_object,
+        columns_to_match, columns_to_add
+      )))
+    },
+    
+    sort = function(by, descending = FALSE, abs_sort = FALSE) {
+      verify_string("by", by, FALSE)
+      verify_bool("descending", descending, FALSE)
+      verify_bool("abs_sort", abs_sort, FALSE)
+      if ((length(descending) > 1) && length(descending) != length(by)) {
+        stop(paste0("'descending' must be the same length as 'by' if more than one entry is supplied. Got 'by' with length ", length(by), " and 'descending' with length ", length(descending), "."))
+      }
+      if ((length(abs_sort) > 1) && length(abs_sort) != length(by)) {
+        stop(paste0("'abs_sort' must be the same length as 'by' if more than one entry is supplied. Got 'by' with length ", length(by), " and 'abs_sort' with length ", length(abs_sort), "."))
+      }
+      return(TableHandle$new(self$.internal_rcpp_object$sort(by, descending, abs_sort)))
+    },
+    
+    merge = function(...) {
+      table_list <- unlist(c(...))
+      if (length(table_list) == 0) {
+        return(TableHandle$new(self$.internal_rcpp_object$merge(NULL)))
```

Comment:
COuldn't this just return self without having to do a round trip to the server?

Reply:
Ehhh.. Right now these R6 objects are set to not be clonable, and I'm not sure I want to return a reference to the same object, if that's possible in R. I can make R6 clonable if the C++ TableHandle has a copy constructor, which I'm sure it does. Not sure what the implications on efficiency there would be, in terms of cloning the whole thing vs a server call.

Reply:
I don't think anything needs to be cloned.  just `return(self)`